instrumentation/rdkafka/lib/opentelemetry/instrumentation/rdkafka/patches/producer.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Rdkafka
+      module Patches
+        # The Producer module contains the instrumentation patch the Producer#produce method
+        module Producer
+          def produce(topic:, payload: nil, key: nil, partition: nil, partition_key: nil, timestamp: nil, headers: nil)
```

Comment:
The specification states that `peer.service` `SHOULD` be set for kafka producers.

https://github.com/open-telemetry/opentelemetry-specification/blob/a1a8676a43dce6a4e447f65518aef8e98784306c/specification/trace/semantic_conventions/messaging.md#apache-kafka

> For Apache Kafka producers, peer.service SHOULD be set to the name of the broker or service the message will be sent to. The service.name of a Consumer's Resource SHOULD match the peer.service of the Producer, when the message is directly passed to another service. If an intermediary broker is present, service.name and peer.service will not be the same.

That being said, i notice we don't do this in our other kafka instrumentation, so i'm not quite sure what the right approach is. I'm ok with merging this as is and then if we need to update the approach for `peer.service` and my related comment for `span.status`, doing that for all kafka instrumentation in a related PR

Reply:
Should this be created as an issue to follow up on?

Reply:
added https://github.com/open-telemetry/opentelemetry-ruby/issues/1047